### ISSUE TITLE
Preview of stl file not working (preview-generator 0.12)

### ISSUE THREAD

**Issue by PhilippeAccorsi** [algoo/preview-generator#163]
## At present
I have this error : 
~~~
18:40:56 INFO  [sqlalchemy.engine.base.Engine:1211][b'uWSGIWorker1Core0'] (1, 1, 'piplus-teeth', '', '.stl', 'application/octet-stream', '{"filename": "piplus-teeth.stl", "content_type": "application/octet-stream", "uploaded_at": "2019-11-21 18:40:56", "file_id": "7468994e-0c8e-11ea-80e ... (25 characters truncated) ... _url": null, "path": "tracim/7468994e-0c8e-11ea-80ef-0242ac120004", "files": ["tracim/7468994e-0c8e-11ea-80ef-0242ac120004"], "depot_name": "tracim"}', '', 'file', 'open', '2019-11-21 18:40:56.601282', '2019-11-21 18:40:56.608331', 0, 0, 0, 'revision', 1, None, 'CONTENT')
KeyError: 'application/octet-stream'
preview_generator.exception.UnsupportedMimeType: Unsupported mimetype: application/octet-stream
KeyError: 'application/octet-stream'
preview_generator.exception.UnsupportedMimeType: Unsupported mimetype: application/octet-stream
~~~

Preview of file not working.

**Comment by PhilippeAccorsi** [comment]
Just delete ".zip" at end of filename
[piplus-teeth.stl.zip](https://github.com/algoo/preview-generator/files/3879780/piplus-teeth.stl.zip)

**Comment by PhilippeAccorsi** [comment]
Test with code in branch Feature/151 better vtk builder

~~~
2019-11-22 16:07:05 INFO  [PreviewGenerator:21][b'uWSGIWorker1Core0'] New Preview builder of class<class 'preview_generator.preview.builder.cad__vtk.ImagePreviewBuilderVtk'>
2019-11-22 16:07:05 INFO  [PreviewGenerator:21][b'uWSGIWorker1Core0'] New Preview builder of class<class 'preview_generator.preview.builder.cad__vtk.ImagePreviewBuilderVtk'>
ERROR: In /work/standalone-x64-build/VTK-source/Rendering/OpenGL2/vtkXOpenGLRenderWindow.cxx, line 445
vtkXOpenGLRenderWindow (0x557993994ff0): bad X server connection. DISPLAY=Fri Nov 22 16:07:07 2019 - Respawned uWSGI worker 2 (new pid: 2010)
Fri Nov 22 16:07:07 2019 - DAMN ! worker 1 (pid: 2007) died, killed by signal 6 :( trying respawn ...
Fri Nov 22 16:07:07 2019 - Respawned uWSGI worker 1 (new pid: 2011)
~~~

**Comment by inkhey** [comment]
> Test with code in branch Feature/151 better vtk builder
> 
> ```
> 2019-11-22 16:07:05 INFO  [PreviewGenerator:21][b'uWSGIWorker1Core0'] New Preview builder of class<class 'preview_generator.preview.builder.cad__vtk.ImagePreviewBuilderVtk'>
> 2019-11-22 16:07:05 INFO  [PreviewGenerator:21][b'uWSGIWorker1Core0'] New Preview builder of class<class 'preview_generator.preview.builder.cad__vtk.ImagePreviewBuilderVtk'>
> ERROR: In /work/standalone-x64-build/VTK-source/Rendering/OpenGL2/vtkXOpenGLRenderWindow.cxx, line 445
> vtkXOpenGLRenderWindow (0x557993994ff0): bad X server connection. DISPLAY=Fri Nov 22 16:07:07 2019 - Respawned uWSGI worker 2 (new pid: 2010)
> Fri Nov 22 16:07:07 2019 - DAMN ! worker 1 (pid: 2007) died, killed by signal 6 :( trying respawn ...
> Fri Nov 22 16:07:07 2019 - Respawned uWSGI worker 1 (new pid: 2011)
> ```

This issue happened because we are in a headless environnement.
check this: https://github.com/vtkiorg/vtki/issues/155

**Comment by inkhey** [comment]
fixed by #167
